src/host/directio.cpp

```diff
@@ -531,61 +529,59 @@ void EventsToUnicode(_Inout_ std::deque<std::unique_ptr<IInputEvent>>& inEvents,
 {
     try
     {
-        std::vector<CHAR_INFO> tempBuffer(buffer.begin(), buffer.end());
-
         const auto size = rectangle.Dimensions();
-        auto tempIter = tempBuffer.cbegin();
         auto outIter = buffer.begin();
 
-        for (til::CoordType i = 0; i < size.Y; i++)
+        for (til::CoordType i = 0; i < size.Y; ++i)
         {
-            for (til::CoordType j = 0; j < size.X; j++)
+            for (til::CoordType j = 0; j < size.X; ++j, ++outIter)
             {
+                auto& in1 = *outIter;
+
+                // If .AsciiChar and .UnicodeChar have the same offset (since they're a union),
+                // we can just write the latter with a byte-sized value to set the former
+                // _and_ simultaneously clear the upper byte of .UnicodeChar to 0. Nice!
+                static_assert(offsetof(CHAR_INFO, Char.AsciiChar) == offsetof(CHAR_INFO, Char.UnicodeChar));

```

Comment:
The other assertion is in the DBCS test file unfortunately. I felt like keeping those assertions close to the code is better since there are alternative ways to write this code (for instance assigning `UnicodeChar` and `AsciiChar` independently).